### ISSUE TITLE
Feedback for the package

### ISSUE THREAD

**Issue by TheGrowingPlant443** [AdvancedR-2021/hmm#6]
### 1. Purpose of the package

I think the purpose of the package is very clear and usefull to have a package that can calculate EM and other algorithms related to Hidden markov moddels and I think it is very nice with a interface that is familiar to lm. 

### 2. Completeness

The package uses the earthquake dataset. It shows how to donwload which is nice but the package is also cable of simulating data it would maybe have been nice to include a simlulated dataset and the earthquake dataset in the package. Over all the package forfills it's goals and it has a nice vignette with alot exsamples on how to simulate, make plots e.g. All Authors a present with contact email. 

### 3. Code quality and sophistication

The package installed easyly but I found a problem with data that was unlikely to have a certain distribution. I made a separate issue for that and it has now been fixed. The package is able to the run it's own exsamples. The fact that alot of the algorthims are implementet in C++ is nice use of high level programming techniques. Including the fact that it has been in the same style as lm is also very nice that one main function that handles everything in the package with is own plot and summary and so on. 

### 4. Documentation
Ever function is documentet with exsamples. The vignette is very well made with theory and exsample on how to do many different things such as defining custom models, simulating data, and how to make plots. Theory is breif and gives a basic understanding of the setup for package. I find the vignette very helpfull and motivating to use the package. There is a short explation of the data but it is not included in the package. 

### 5. Interface
The interface well made but making custom functions for the HMM mixture is a bit difficult to use. The arguments like delta and gamma is named after the variables in the litterature which makes it easy to understand but the remaining variables like lls, lls_mle rdist would be nice to have a more saying name. The packages does have error handling and they catch alot of mistakes and gives good feedback to user on how to fix it. 

### 7. Conclusion
In conclusion the package very professionally made with one main function. It installs and works with exsamples and has a well made documentation. I really like that the algorithms were implementet in C++ and that the package was made with one main functions that uses all other functions. In the final version I would like to see a dataset included with the package and it would be nice to have futher functions like forecasting. Maybe if possible a more userfriendly way to make a custom function for the HMM.
